fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -819,9 +819,15 @@ std::set<Tag> CommitBatchContext::getWrittenTagsPreResolution() {
 		// empty sets are sent to all logs.
 		return transactionTags;
 	}
+	if (pProxyCommitData->idempotencyClears.size()) {
```

Comment:
From the reply below:

> because the actor that enqueues an item to be cleared may be scheduled between pre and post resolution

This check is done pre-resolution - so this condition is not always guaranteed to be true for versions with idempotent transactions?

Does this imply that there is no reliable way to detect versions with idempotent transactions before pre-resolution?

Reply:
per our conversation, I will add an assert for case 1 and add a comment for case 2 to clarify the different situations.